metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -421,34 +430,52 @@ def get_workunits(self) -> Iterable[WorkUnit]:
         if self.config.include_usage_statistics:
             yield from self.usage_extractor.report_usage_stat()
 
-        # Clean up stale entities if configured.
-        yield from self.stale_entity_removal_handler.gen_removed_entity_workunits()
+        if self.config.profiling.enabled:
+            yield from self.profiler.get_workunits(self.db_tables)
+        if self.is_stateful_ingestion_configured():
+            # Clean up stale entities.
+            yield from self.stale_entity_removal_handler.gen_removed_entity_workunits()
 
     def _process_project(
         self, conn: bigquery.Client, bigquery_project: BigqueryProject
     ) -> Iterable[MetadataWorkUnit]:
         project_id = bigquery_project.id
 
+        self.db_tables[project_id] = {}
+        self.db_views[project_id] = {}
+
         database_workunits = self.gen_project_id_containers(project_id)
 
         for wu in database_workunits:
             self.report.report_workunit(wu)
             yield wu
 
-        bigquery_project.datasets = BigQueryDataDictionary.get_datasets_for_project_id(
-            conn, project_id
-        )
+        try:
+
+            bigquery_project.datasets = (
+                BigQueryDataDictionary.get_datasets_for_project_id(conn, project_id)
+            )
+        except Exception as e:
+            logger.error(
+                f"Unable to get datasets for project {project_id}, skipping. The error was: {e}"
+            )
+            return None
 
         for bigquery_dataset in bigquery_project.datasets:
 
             if not self.config.dataset_pattern.allowed(bigquery_dataset.name):
                 self.report.report_dropped(f"{bigquery_dataset.name}.*")
                 continue
+            try:
+                yield from self._process_schema(conn, project_id, bigquery_dataset)
+            except Exception as e:
+                logger.error(
+                    f"Unable to get tables for dataset {bigquery_dataset.name} in project {project_id}, skipping. The error was: {e}"
+                )
+                continue
 
-            yield from self._process_schema(conn, project_id, bigquery_dataset)
-
-        if self.config.include_usage_statistics:
-            yield from self.usage_extractor.generate_usage_for_project(project_id)
+            if self.config.include_usage_statistics:
+                yield from self.usage_extractor.generate_usage_for_project(project_id)
```

Comment:
shouldn't this be outside the for loop?

Reply:
@treff7es, is this not going to be an issue? This will be invoked each time we process a dataset for the entire project.